libqtile/widget/groupbox.py

```diff
@@ -347,6 +352,10 @@ def draw(self):
 
         offset = self.margin_x
         for i, g in enumerate(self.groups):
+
+            if self.show_screen_only and self.qtile.current_screen != self.bar.screen:
```

Comment:
I'm confused. This test doesn't look up anything to do with the groups that you're looping over so it's either true or false for all groups.

What are you actually trying to do?